camayoc/tests/qcs/test_profiles.py

```diff
@@ -61,47 +67,96 @@ def test_update(shared_client, cleanup, scan_host):
            the data
     :expectedresults: The newtork profile entry is created and updated.
     """
-    cred = HostCredential(client=shared_client, password=str(uuid4()))
+    cred = HostCredential(client=shared_client, password=uuid4())
     cred.create()
+    cleanup.append(cred)
     profile = NetworkProfile(
         client=shared_client,
         hosts=[scan_host],
         credential_ids=[cred._id],
     )
     profile.create()
+    cleanup.append(profile)
     assert_matches_server(profile)
     profile.hosts.append('0.0.0.0')
-    cred2 = HostCredential(password=str(uuid4()))
+    cred2 = HostCredential(password=uuid4())
     cred2.create()
+    cleanup.append(cred2)
     profile.credentials.append(cred2._id)
     profile.update()
-    cleanup.extend([cred, cred2, profile])
     assert_matches_server(profile)
 
 
-@pytest.mark.skip
-def test_negative_update_to_invalid(cleanup):
+@pytest.mark.parametrize('scan_host', CREATE_DATA)
+def test_update_bad_credential(scan_host, cleanup):
     """Attempt to update valid profile with invalid data.
 
     :id: 79954c63-608c-46b3-81eb-e2a1e984473e
     :description: Create valid network profile, and then attempt to update
-        it with invalid data.
+        it with an invalid credential.
+    :steps:
+        1) Create valid credential and profile
+        2) Update the profile with invalid credentials
+    :expectedresults: Error codes are returned and the profile is not updated.
+    :caseautomation: automated
+    """
+    profile = gen_valid_profile(cleanup, api.Client(), scan_host)
+    original_data = copy.deepcopy(profile.fields())
+
+    # Case "a" add credential that doesnt exist
+    # The server never assigns negative values
+    badcred = {'id': -1}
+    profile.credentials.append(badcred['id'])
+    assert_profile_update_fails(original_data, profile)
+
+
+@pytest.mark.parametrize('data', ['credentials', 'hosts'])
+def test_update_remove_data(cleanup, data):
+    """Attempt to update valid profile with either no hosts or no credentials.
+
+    :id: 49feb858-319f-4f77-b330-65426dfd1734
+    :description: Create valid network profile, and then attempt to update
+        it to have either no credentials or no hosts.
     :steps:
         1) Create valid credential and profile
-        2) Update the profile with:
-            a) invalid credentials
-            b) Remove all credentials
-            c) Remove all hosts
-            d) An invalid host
-            e) Non-existent fields
+        2) Update the profile with no credentials or no hosts
```

Comment:
So, as you can see, I ended up organizing these test scenarios differently. 
* Case b) and c) I implemented in `test_update_remove_data` because I could parametrize them and otherwise use the same code. 
* Case a) is in `test_update_bad_credential`
*  case d) is implemented with `test_update_with_bad_host` 
* case e) is still up for debate in [issue 205](https://github.com/quipucords/quipucords/issues/205) so I omitted it

Reply:
Sounds good